### PR TITLE
[Merged by Bors] - feat: Complete `NNReal` coercion lemmas

### PR DESCRIPTION
**Pull request by YaelDillies**
Add a few missing lemmas about the coercion `NNReal → Real`. Remove a bunch of `protected` on the existing coercion lemmas (so that it matches the convention for other coercions). Rename `NNReal.coe_eq` to `NNReal.coe_inj`

From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Real/NNReal.lean [hunk 1]
```diff
@@ -163,19 +163,17 @@ example : Nontrivial ℝ≥0 := by infer_instance
 protected theorem coe_injective : Injective ((↑) : ℝ≥0 → ℝ) := Subtype.coe_injective
 #align nnreal.coe_injective NNReal.coe_injective
 
-@[simp, norm_cast]
-protected theorem coe_eq {r₁ r₂ : ℝ≥0} : (r₁ : ℝ) = r₂ ↔ r₁ = r₂ :=
+@[simp, norm_cast] lemma coe_inj {r₁ r₂ : ℝ≥0} : (r₁ : ℝ) = r₂ ↔ r₁ = r₂ :=
   NNReal.coe_injective.eq_iff
-#align nnreal.coe_eq NNReal.coe_eq
+#align coe_inj NNReal.coe_inj
```
Comment (urkud) [inline Mathlib/Data/Real/NNReal.lean:168]: Why do you drop the namespace here?

File: Mathlib/Data/Real/NNReal.lean [hunk 2]
```diff
@@ -376,26 +369,25 @@ protected theorem coe_ofNat (n : ℕ) [n.AtLeastTwo] :
...
+@[simp, norm_cast] lemma coe_le_one : (r : ℝ) ≤ 1 ↔ r ≤ 1 := by rw [← coe_le_coe, coe_one]
+@[simp, norm_cast] lemma coe_lt_one : (r : ℝ) < 1 ↔ r < 1 := by rw [← coe_lt_coe, coe_one]
+
+@[mono] lemma coe_mono : Monotone ((↑) : ℝ≥0 → ℝ) := fun _ _ => NNReal.coe_le_coe.2
 #align nnreal.coe_mono NNReal.coe_mono
 
-@[gcongr]
-theorem _root_.NNReal.toReal_le_toReal {x y : NNReal} (h : x ≤ y) : (x : ℝ) ≤ y := NNReal.coe_mono h
+/-- Alias for the use of `gcongr` -/
+@[gcongr] alias ⟨_, GCongr.toReal_le_toReal⟩ := coe_le_coe
```
Comment (urkud) [inline Mathlib/Data/Real/NNReal.lean:390]: Should it be `_root_.GCongr.nnrealToReal_le_toReal`? I'm not sure.

File: Mathlib/Data/Real/NNReal.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Data/Real/NNReal.lean:168]:
```suggestion
#align nnreal.coe_inj NNReal.coe_inj
```

**Comment by urkud** [comment]
Otherwise LGTM. Thanks!
bors d+

File: Mathlib/Data/Real/NNReal.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Data/Real/NNReal.lean:168]: Whoops, accident indeed

File: Mathlib/Data/Real/NNReal.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Data/Real/NNReal.lean:390]: Don't think it really matters at this point. We can wait a bit longer and make a mathlib-wide decision.

**Pull request closed without merging**